atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -0,0 +1,263 @@
+        parser = DockerfileParser(df_path)
+        current_release = parser.labels[label_key]
+        next_release = self.get_next_release(current_release)
+        self.log.info("New Release: %s", next_release)
+        self.bump_local_file(parser, label_key, next_release)
+
+        # Stage it
+        index = repo.index
+        index.add(os.path.basename(df_path))
+
+        # Commit the change
+        author = Signature(self.author_name, self.author_email)
+        committer = Signature(self.committer_name, self.committer_email)
+        repo.create_commit(branch.name, author, committer, self.commit_message,
+                           index.write_tree(), [repo.head.peel().hex])
+
+        # Push it
+        self.log.info("Pushing to git repository")
+        ssh_command = '/usr/bin/ssh -o StrictHostKeyChecking=no'
+        os.environ['GIT_SSH_COMMAND'] = ssh_command
+
+        # Why doesn't this work?
+        #   remote.push([branch.name])
+        # Instead, run the git command to do it
+        cmd = ['/usr/bin/git', 'push', 'origin']
+        try:
+            output = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
+                                             cwd=repo.workdir)
+            self.log.debug("git (success): %s", output)
+        except subprocess.CalledProcessError as exc:
+            self.log.error("git (%d): %s", exc.returncode, exc.output)
+            raise
+        except AttributeError:
+            # py2.6
+            with open('/dev/null', 'r+') as devnull:
+                p = subprocess.Popen(cmd,
+                                     stdin=devnull,
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.STDOUT,
+                                     cwd=repo.workdir)
+                (output, empty) = p.communicate()
+                status = p.wait()
+                if status != 0:
+                    self.log.error("git (%d): %s", status, output)
+                    raise RuntimeError("exit code %d" % status)
+
+                self.log.debug("git (success): %s", output)
+
+    def verify_branch(self, branch):
+        commit = branch.target.hex
+        if commit != self.git_ref:
+            raise RuntimeError("Branch '%s' is at commit %s (expected %s)" %
+                               (branch.shorthand, commit, self.git_ref))
+
+        self.log.info("Branch '%s' is at expected commit (%s)" %
+                      (branch.shorthand, self.git_ref))
+
+    def run(self):
+        """
+        run the plugin
+        """
+
+        if self.workflow.build_process_failed:
+            self.log.info("Build already failed, not incrementing release")
+            return
+
+        # Ensure we can use the git repository already checked out for us
+        source = self.workflow.source
+        assert isinstance(source, GitSource)
+        repo = init_repository(source.get())
+        branch = repo.lookup_branch(source.git_commit)
```

Comment:
What if user passed an actual commit instead of a branch name as `git_commit`? That can theoretically happen, AFAICS. (Isn't that what Koji will do? https://github.com/release-engineering/koji-containerbuild/blob/master/koji_containerbuild/plugins/builder_containerbuild.py#L335)